service/src/integrationTest/kotlin/fi/espoo/evaka/invoicing/FeeDecisionIntegrationTest.kt

```diff
@@ -942,9 +948,47 @@ class FeeDecisionIntegrationTest : FullApplicationTest() {
 
         val (_, response, _) = http.post("/decisions/confirm")
             .asUser(user)
-            .jsonBody(objectMapper.writeValueAsString(draftWithFutureDates.id))
+            .jsonBody(objectMapper.writeValueAsString(listOf(draftWithFutureDates.id)))
             .response()
         assertEquals(400, response.statusCode)
+        assertEquals(
+            "feeDecisions.confirmation.tooFarInFuture",
+            objectMapper.readTree(response.body().asString("text/json")).get("errorCode").textValue()
+        )
+    }
+
+    @Test
+    fun `confirmDrafts when decision at last possible confirmation date exists`() {
+        doReturn(1L).whenever(evakaEnv).nrOfDaysFeeDecisionCanBeSentInAdvance
+        val now = HelsinkiDateTime.now()
+        val clock = MockEvakaClock(now)
+        val draftWithFutureDates = testDecisions.find { it.status == FeeDecisionStatus.DRAFT }!!
+            .copy(validDuring = DateRange(clock.today().plusDays(1), clock.today().plusYears(1)))
```

Comment:
You don't need `clock` here, you could just use `now.toLocalDate()` instead of `clock.today()`.